llvm/include/llvm/Analysis/CtxProfAnalysis.h

```diff
@@ -35,6 +51,20 @@ class PGOContextualProfile {
     return *Profiles;
   }
 
+  bool isFunctionKnown(const Function &F) const {
+    return getDefinedFunctionGUID(F) != 0;
+  }
+
+  uint32_t allocateNextCounterIndex(const Function &F) {
+    assert(isFunctionKnown(F));
+    return FuncInfo.find(getDefinedFunctionGUID(F))->second.NextCounterIndex++;
```

Comment:
Is this performing the lookup 2x on an assert build? If so maybe rewrite this as 
```
auto& [found, iter] = FuncInfo.find(....);
(void) found;
assert(found && "function not found");
```

Reply:
ya, but I wanted the code to be more readable - i.e. the assertion is that the function is known rather than its implementation.